pkg/cvo/cvo.go

```diff
@@ -163,6 +165,10 @@ func New(
 	}
 
 	cvInformer.Informer().AddEventHandler(optr.eventHandler())
+	proxyInformer.Informer().AddEventHandler(optr.eventHandler())
+
+	optr.proxyLister = proxyInformer.Lister()
+	optr.cacheSynced = append(optr.cacheSynced, proxyInformer.Informer().HasSynced)
```

Comment:
4.2 the type exists for proxy.
but 4.1 the type doesn't exit for proxy.

So when user goes from 4.1.z to 4.2.z ; the CVO updates itself to a version that expects proxy type to exist, but the proxy type will be created by the cluster-config-operator. So this cache will never sync and CVO will be stuck.

So it's important our changes keep in mind that the type might not exist.